tools/offline_log_viewer/storage.py

```diff
@@ -215,6 +215,11 @@ def from_stream(f, index):
                 return None
             assert len(data) == records_size
             return Batch(index, header, data)
+
+        if len(data) < HEADER_SIZE:
+            # Short read, probably log being actively written or unclean shutdown
+            return None
+
         assert len(data) == 0
```

Comment:
seems like `assert len(data) == 0` is no longer possible. indeed, the if statement also doesn't seem necessary now since data will definitely be < header size.